src/test/java/org/myrobotlab/service/HttpClientTest.java

```diff
@@ -0,0 +1,99 @@
+package org.myrobotlab.service;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Before;
+import org.myrobotlab.framework.Message;
+import org.myrobotlab.framework.Service;
+import org.myrobotlab.service.data.HttpData;
+
+public class HttpClientTest extends
+    AbstractServiceTest /* implements HttpDataListener - not useful .. "yet" */ {
+
+  TestCatcher catcher;
+
+  @Before
+  public void setUp() throws Exception {
+    catcher = (TestCatcher) Runtime.start("catcher", "TestCatcher");
+  }
+
+  @Override
+  public Service createService() {
+    return (Service) Runtime.start("http", "HttpClient");
+  }
+
+  /**
+   * publishHttpData contains more information content type, response code,
+   * etc... need to subscribe to it manually for testing purposes
+   * 
+   */
+  /*
+   * @Override //FIXME - add blocking queue - to verify public void
+   * onHttpData(HttpData data) { log.info(data.toString()); queue.add(data); }
+   */
+
+  /**
+   * for testing purposes
+   * 
+   */
+  /*
+   * @Override //FIXME - add blocking queue - to verify public void
+   * onHttpResponse(String data) { log.info(data); queue.add(data); }
+   */
+
+  @Override
+  public void testService() throws Exception {
+    HttpClient http = (HttpClient) Runtime.start("http", "HttpClient");
+
+    // this is how a listener might subscribe
+    // TODO - put dynamically subscribing into framework
+    // with interface inspection ??
+
+    // FIXME - add make the attach !
+    http.attach(catcher);
+
+    catcher.clear();
+    String data = http.get("https://postman-echo.com/get?foo1=bar1&foo2=bar2");
```

Comment:
this test requires an internet connection to pass making it unreliable.

Reply:
fixed